src/main/java/org/prebid/server/bidder/ix/IxAdapter.java

```diff
@@ -64,19 +77,34 @@ private static void validatePreBidRequest(PreBidRequest preBidRequest) {
         }
     }
 
-    private BidRequest createBidRequest(AdapterRequest adapterRequest, PreBidRequestContext preBidRequestContext) {
-        final List<AdUnitBid> adUnitBids = adapterRequest.getAdUnitBids();
-
-        validateAdUnitBidsMediaTypes(adUnitBids, ALLOWED_MEDIA_TYPES);
+    private List<BidRequest> makeRequests(List<AdUnitBid> adUnitBids, PreBidRequestContext preBidRequestContext) {
+        final List<BidRequest> prioritizedRequests = new ArrayList<>();
+        final List<BidRequest> regularRequests = new ArrayList<>();
+
+        for (AdUnitBid adUnitBid : adUnitBids) {
+            final List<Format> sizes = adUnitBid.getSizes();
+            for (Format size : sizes) {
+                final BidRequest bidRequest = createBidRequest(adUnitBid, size, preBidRequestContext);
+                // prioritize slots over sizes
+                if (sizes.indexOf(size) == 0) {
+                    prioritizedRequests.add(bidRequest);
+                } else {
+                    regularRequests.add(bidRequest);
+                }
+            }
+        }
+        return Stream.concat(prioritizedRequests.stream(), regularRequests.stream())
+                // cap the number of requests to requestLimit
+                .limit(REQUEST_LIMIT)
+                .collect(Collectors.toList());
+    }
 
-        final List<Imp> imps = makeImps(adUnitBids, preBidRequestContext);
+    private BidRequest createBidRequest(AdUnitBid adUnitBid, Format size, PreBidRequestContext preBidRequestContext) {
+        final IxParams ixParams = parseAndValidateParams(adUnitBid);
```

Comment:
Move pls parsing params to `for (AdUnitBid adUnitBid : adUnitBids) {...}` loop.
In other case we'll do parsing on each adUnitBid/size iteration.